src/redux/slices/news/middleware.ts

```diff
@@ -56,14 +60,74 @@ const getIndividualNewsById = (id: string) => async (dispatch: AppDispatch) => {
   }
 }
 
+const createNews = (formData: ICreateNewsProps) => async (dispatch: AppDispatch) => {
+  try {
+    await API.news.createNews(formData)
+
+    dispatch(setCreateNewsSubmitSuccess(true))
+
+    dispatch(setError(null))
+  } catch (error) {
+    dispatch(setError((error as IError).response?.data?.status?.message))
+  }
+}
+
+const changeNews = (formData: IChangeNewsFormProps) => async (dispatch: AppDispatch) => {
+  try {
+    await API.news.changeNews(formData)
+
+    dispatch(setChangeNewsSubmitSuccess(true))
+
+    dispatch(setError(null))
+  } catch (error) {
+    dispatch(setError((error as IError).response?.data?.status?.message))
+  }
+}
+
 const clearNewsList = () => async (dispatch: AppDispatch) => {
   dispatch(setNewsList([]))
   dispatch(setPageSize(0))
   dispatch(setTotalItems(0))
 }
 
+const resetCreateNewsSubmitSuccess = () => (dispatch: AppDispatch) => {
+  dispatch(setCreateNewsSubmitSuccess(false))
+}
+
+const resetChangeNewsSubmitSuccess = () => (dispatch: AppDispatch) => {
+  dispatch(setChangeNewsSubmitSuccess(false))
+}
+
+const deleteIndividualNews = (uuid: string) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setDeleteNewsLoading(true))
+
+    await API.news.deleteIndividualNews(uuid)
+
+    const reqBody = {
+      page: 1,
+    }
+    const response = await API.news.getNews(reqBody)
+
```

Comment:
You should reset then get as it will cause data issue

Reply:
done

Reply:
Use middleware

Reply:
done